packages/babel-plugin-component-annotate/src/index.ts

```diff
@@ -443,18 +435,109 @@ function attributeNamesFromState(state: AnnotationPluginPass): [string, string,
   return [webComponentName, webElementName, webSourceFileName];
 }
 
-function isReactFragment(t: typeof Babel.types, openingElement: Babel.NodePath): boolean {
+function collectFragmentContext(programPath: Babel.NodePath): FragmentContext {
+  const fragmentAliases = new Set<string>();
+  const reactNamespaceAliases = new Set<string>(["React"]); // Default React namespace
+
+  programPath.traverse({
+    ImportDeclaration(importPath) {
+      const source = importPath.node.source.value;
+
+      // Handle React imports
+      if (source === "react" || source === "React") {
+        importPath.node.specifiers.forEach((spec) => {
+          if (spec.type === "ImportSpecifier" && spec.imported.type === "Identifier") {
+            // import { Fragment } from 'react' -> Fragment
+            // import { Fragment as F } from 'react' -> F
+            if (spec.imported.name === "Fragment") {
```

Comment:
Another question, should we safeguard that it's also
`spec.source.value === "react"` so we know Fragment is actually from react and not some user code?

Reply:
I think this should be covered with the outer check [`if (source === "react" || source === "React")`](https://github.com/getsentry/sentry-javascript-bundler-plugins/pull/767/files/4063095b0a200c6ce81bc3fef47ecc8dce4279d2#diff-546abc101fe501127b609cbca13f95a13192ce03abce53ecbd9f37db97166ffbR447)